sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -275,6 +275,10 @@ static BoundValue fromDefault(@Nullable Object value) {
    * @return An object that implements the interface {@code <T>}.
    */
   <T extends PipelineOptions> T as(Class<T> iface) {
+    return as(iface, null);
+  }
+
+  <T extends PipelineOptions> T as(Class<T> iface, PipelineOptions pipelineOptions) {
```

Comment:
Is the only purpose of passing `pipelineOptions` to tell whether `InvokePipelineOptionsFactory` should be called? What are the cases when `pipelienOptions` is null?

Reply:
there're two place 1. calling xxPipelineOptions.as(pipelineOptions.class) to return a another type of pipelineOptions. 2. creating the a raw pipelineOptions using PipelineOptionsFactory.create(). 
#1 is where we want to initialize the generator.
and when execute #2 initialize the generator is already wired in.